### PR TITLE
COMMUNITY-100 | Create new CI pipelines for Jenkins

### PR DESCRIPTION
**Pull request by ManuelJNunez**
This PR is to add new pipelines to validate connector-py using Jenkins.
The build_and_test pipeline will do the following:
1. Download connext libs.
2. Run tests on Python 2.7 and 3.6.
3. Publish package to our internal pypi server if a tag with format X.Y.Z-dev is pushed (this package can be overwritten without any problem).

The build_doc pipeline will do the following:
1. Download connext libs (it sounds weird but it seems to be necessary).
2. Install python dependencies.
3. Build doc and store the artifacts on the pipeline.
4. Publish doc if a tag with format X.Y.Z-doc is pushed.

Keep in mind that this connector is in maintenance not in development, so we should not work too much on the pipeline.

### CODE REVIEW

File: resources/jenkins/build_and_test.groovy [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * works of the software solely for use with RTI Connext DDS. Licensee may
+ * redistribute copies of the software provided that all such copies are subject
+ * to this license. The software is provided "as is", with no warranty of any
+ * type, including any warranty for fitness for any purpose. RTI is under no
+ * obligation to maintain or support the software. RTI shall not be liable for
+ * any incidental or consequential damages arising out of the use or inability
+ * to use the software.
+ */
+
+@Library("rticommunity-jenkins-pipelines@feature/COMMUNITY-95") _
```
Comment (ManuelJNunez) [inline resources/jenkins/build_and_test.groovy:13]: - [ ] Remove before merging

File: resources/jenkins/build_doc.groovy [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ * works of the software solely for use with RTI Connext DDS. Licensee may
+ * redistribute copies of the software provided that all such copies are subject
+ * to this license. The software is provided "as is", with no warranty of any
+ * type, including any warranty for fitness for any purpose. RTI is under no
+ * obligation to maintain or support the software. RTI shall not be liable for
+ * any incidental or consequential damages arising out of the use or inability
+ * to use the software.
+ */
+
+@Library("rticommunity-jenkins-pipelines@feature/COMMUNITY-95") _
```
Comment (ManuelJNunez) [inline resources/jenkins/build_doc.groovy:13]: - [ ] Remove before merging

File: test/python/test_rticonnextdds_connector.py [hunk 3]
```diff
@@ -124,10 +124,11 @@ def test_get_version(self):
         # - the build ID of lua_binding.1.0
         # Each build ID can be of the form X.X.X[.X], where the last integer
         # is not always present
+        version_regex = "([0-9]\\.){2,3}[0-9]{0,2}_[0-9]{8}T[0-9]{6}Z"
         assert bool(re.match("RTI Connector for Python, version (([0-9]\\.){2}[0-9]|unknown)", version_string, re.DOTALL)) == True
-        assert bool(re.match(".*NDDSCORE_BUILD_([0-9]\\.){2,3}[0-9]_[0-9]{8}T[0-9]{6}Z", version_string, re.DOTALL)) == True
-        assert bool(re.match(".*NDDSC_BUILD_([0-9]\\.){2,3}[0-9]_[0-9]{8}T[0-9]{6}Z", version_string, re.DOTALL)) == True
-        assert bool(re.match(".*RTICONNECTOR_BUILD_([0-9]\\.){2,3}[0-9]_[0-9]{8}T[0-9]{6}Z", version_string, re.DOTALL)) == True
+        assert bool(re.match(".*NDDSCORE_BUILD_" + version_regex, version_string, re.DOTALL)) == True
```
Comment (ManuelJNunez) [inline test/python/test_rticonnextdds_connector.py:129]: This was extracted from release/connector/1.2.3

Comment (Alxe) [reply test/python/test_rticonnextdds_connector.py:129]: We should be using `r''` strings but I don't think you should be the one making that change. :)

**Review by Alxe — APPROVED** [review]

**Pull request merged**